src/protocol/relayer.md

```diff
@@ -0,0 +1,59 @@
+# Layer 1 Relayer/Bridge Protocol
+
+The Fuel relayer/bridge protocol is a set of rules that govern the interaction between the Fuel blockchain and the
+Layer 1 (L1) blockchain (e.g. Ethereum).  
+
+The Fuel blockchain can emit messages that will be processed by the smart contract on the L1 blockchain. The smart
+contract on the L1 can also emit events that will be processed by the Fuel blockchain.
+This is used to move any data between the L1 blockchain and the Fuel blockchain.
+
+## Fuel Message Outbox
+
+The message outbox is the set of messages sent to the L1 blockchain from the Fuel blockchain.
+
+## Fuel Event Inbox
+
+The event inbox is the set of events received from the L1 blockchain by the Fuel blockchain.
+
+The block producer will receive the list of events from the L1 by some relayer, and then include both the count and the
+merkle root of the events in the block header.
+
+There are two types of events that can be received from the L1:
+
+1. Messages
+2. Transactions
+
+### Messages
+
+An arbitrary message sent from the L1 to the Fuel blockchain. This can be used to move assets from the L1
+to the Fuel blockchain or send other arbitrary information to the Fuel blockchain.
+
+| name        | type    | description                                                         |
+|-------------|---------|---------------------------------------------------------------------|
+| `sender`    | `bytes[32]` | The identity of the sender of the message on the L1                 |
+| `recipient` | `bytes[32]` | The recipient of the message on the Fuel Blockchain                 |
+| `nonce`     | `bytes[32]` | Unique identifier of the message assigned by the L1 contract                                 |
+| `amount`    | `uint64`  | The amount of the base asset transfer                              |
+| `data`      | `byte[]`  | Arbitrary message data                                              |
+
+### Transactions
+
+These are transactions that are submitted on the L1 that must be executed on the Fuel blockchain.
+This "Forced Transaction Inclusion" is a security feature that allows participants of the Fuel Blockchain to access
+their funds in the (unlikely) event that the Fuel blockchain block production is compromised or malicious.
```

Comment:
Could mention censorship-resistancy here, as that's the primary technical form of compromise this protects against